internal/credential/subtype.go

```diff
@@ -2,18 +2,17 @@ package credential
 
 import (
 	"strings"
-
-	"github.com/hashicorp/boundary/internal/credential/vault"
+	"sync"
 )
 
-type SubType int
+type Subtype int
 
 const (
-	UnknownSubtype SubType = iota
+	UnknownSubtype Subtype = iota
 	VaultSubtype
```

Comment:
What would be in that package and what would stay in this package?

Reply:
That package would contain details on cross boundary go types and interfaces (Subtype, Entity, Aggregate, Resource, etc...) and common functions that operate on them.  In this package it would be types and interfaces specific to the credential domain (Store, Library, Credential, Dynamic, etc...) and the functions that operate on them.

Reply:
I like that idea